### PR TITLE
Introduce DEAL_II_VERSION_LTE

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Comment by peterrum** [comment]
@drwells Yes!

**Comment by drwells** [comment]
Its a little redundant, but why not? It makes more sense in some cases.

**Review by bangerth — APPROVED** [review]

File: include/deal.II/base/config.h.in [hunk 1]
```diff
@@ -268,6 +268,19 @@
     >=  \
     (major)*10000 + (minor)*100 + (subminor))
 
+/**
+ * Returns true if the used deal.II version is less or equal than the
```
Comment (bangerth) [inline include/deal.II/base/config.h.in:272]:
```suggestion
 * Returns true if the used deal.II version is less than or equal the 
```

Comment (peterrum) [reply include/deal.II/base/config.h.in]: Done!

**Comment by tjhei** [comment]
I am not sure how this helps because it breaks if we do another point release like 9.3.4. I think you would need to ignore the subminor for it to make any sense.
I think the existing macro does the job just fine. The mistake was to use 10.0 instead of 9.4.

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
But I agree with @tjhei that the subminor needs to be set sufficiently high (as I would have done intuitively) or ignored.

**Comment by tjhei** [comment]
So then what is the reason to have
```
#if DEAL_II_VERSION_LTE(9, 3, 99)
A
#else
B
#endif
```
vs

```
#if DEAL_II_VERSION_GTE(9, 4, 0)
B
#else
a
#endif
```

I would think the latter is a more sensible choice.

**Comment by bangerth** [comment]
I think what @tjhei is saying is that you should just patch your own projects with something like https://github.com/geodynamics/aspect/pull/4520 . This is backward compatible and avoids the awkward issue of the subminor.

**Comment by kronbichler** [comment]
How do we want to proceed here?

**Comment by peterrum** [comment]
I haven't thought about point releases. I am fine to close the PR. But we should agree for the future that we don't decide in advance (one year before the actual release!) that we will create a major release. Bumping form minor to major release can still be decided on shortly before the actual release.

**Comment by tjhei** [comment]
> I am fine to close the PR.

I agree that his is the best choice.

>  But we should agree for the future that we don't decide in advance

Yes, we made two mistakes:
1. Bump the version to 10.0 instead of 9.4
2. Use GTE(10,0,0) instead of GTE(9,4,0)

It is always safer just to bump the minor when using the GTE macro...

**Pull request closed without merging**